database/mongodb/mongodb.go

```diff
@@ -217,12 +217,10 @@ func parseInt(urlParam string, defaultValue int) (int, error) {
 }
 func (m *Mongo) SetVersion(version int, dirty bool) error {
 	migrationsCollection := m.db.Collection(m.config.MigrationsCollection)
-	if err := migrationsCollection.Drop(context.TODO()); err != nil {
-		return &database.Error{OrigErr: err, Err: "drop migrations collection failed"}
-	}
-	_, err := migrationsCollection.InsertOne(context.TODO(), bson.M{"version": version, "dirty": dirty})
+	opts := options.Update().SetUpsert(true)
+	_, err := migrationsCollection.UpdateOne(context.TODO(), bson.M{}, bson.D{{Key: "$set", Value: bson.M{"version": version, "dirty": dirty}}}, opts)
```

Comment:
Will this work with existing DBs? e.g. how does this update existing entries?
Right now, there's only ever one document in the migrations collection.

Reply:
Yes @dhui, it is working with existing entries in DB. To test this behaviour, I have added some sleep between the SetVersion functions in migrate.js and monitored changes in the document. It updated the dirty flag for the existing entry and for the new version it replaced the document with a new document.  When the migration ended the collection contained only a single document.